docs/internals/contributing/writing-code/submitting-patches.txt

```diff
@@ -201,17 +201,22 @@ level:
     class MyDeprecatedTests(unittest.TestCase):
         ...
 
-You can also add a test for the deprecation warning::
+You should also add a test for the deprecation warning::
 
     from django.utils.deprecation import RemovedInDjangoXXWarning
 
 
+    # RemovedInDjangoXXWarning.
     def test_foo_deprecation_warning(self):
         msg = "Expected deprecation message"
         with self.assertWarnsMessage(RemovedInDjangoXXWarning, msg):
             # invoke deprecated behavior
             ...
 
+You should also add a ``RemovedInDjangoXXWarning`` comment above any code which
+will need to be removed or altered for the XX release. This makes it easier to
+remove when deprecation ends.
```

Comment:
These kind of comments are not necessary for all code that will be removed when a deprecation ends, it is especially unnecessary for tests of deprecation warnings or tests decorated with `@ignore_warnings()`.

We're trying to add them to hooks added to keep the previous behavior, or standalone items that are unnecessary or unused when a deprecation ends.

Reply:
Right you search for `RemovedInDjangoXXWarning` and need to see everything, so the places with warnings already show up, it's the more subtle stuff :+1: